server/vulnerabilities/cve.go

```diff
@@ -148,9 +181,38 @@ func checkCVEs(ctx context.Context, ds fleet.Datastore, logger kitlog.Logger, cp
 							}
 							matchingCPEs = append(matchingCPEs, cpe)
 						}
-						err = ds.InsertCVEForCPE(ctx, matches.CVE.ID(), matchingCPEs)
+
+						newCount, err := ds.InsertCVEForCPE(ctx, matches.CVE.ID(), matchingCPEs)
 						if err != nil {
 							level.Error(logger).Log("cpe processing", "error", "err", err)
+							continue // do not report a recent vuln that failed to be inserted in the DB
+						}
+
+						// collect as recent vuln only if newCount > 0, otherwise we would send
+						// webhook requests for the same vulnerability over and over again until
+						// it is older than 2 days.
+						if collectVulns && newCount > 0 {
+							vuln, ok := matches.CVE.(*feednvd.Vuln)
+							if !ok {
+								level.Error(logger).Log("recent vuln", "unexpected type for Vuln interface", "type", fmt.Sprintf("%T", matches.CVE))
+								continue
+							}
+
+							if rawPubDate := vuln.Schema().PublishedDate; rawPubDate != "" {
```

Comment:
Nit: Invert logic to decrease indentation in this method (having a hard time reviewing with the horizontal scroll :) (and add a log)
```go
rawPubDate := vuln.Schema().PublishedDate
if rawPubDate == "" {
  level.Error(logger).Log("vuln", someStringRepresentationOf(vuln), "err", "empty published")
  continue
}
```